KANOPY-1192 upgrade helm version to 3.18.6

rojanDinc opened the pull request:
### Description

This PR upgrades the helm base image version that we use for this plugin to helm v3.18.6.

### Testing

`make test` of this package passing

```
$ make test
golangci-lint run --timeout=5m ./...
0 issues.
go test -cover -run=.* ./...
        github.com/kanopy-platform/drone-helm3/cmd/drone-helm           coverage: 0.0% of statements
ok      github.com/kanopy-platform/drone-helm3/internal/env     (cached)        coverage: 84.8% of statements
ok      github.com/kanopy-platform/drone-helm3/internal/helm    (cached)        coverage: 89.8% of statements
ok      github.com/kanopy-platform/drone-helm3/internal/run     (cached)        coverage: 77.2% of statements
```

Image build passing in kanopy-container-images: https://drone.corp.mongodb.com/10gen/kanopy-container-images/549

Deployment passing in `drone-e2e-test`: https://drone.test.corp.mongodb.com/10gen/drone-e2e-test/1087/1/2

rojanDinc committed:
upgrade helm version used for the docker image

jaswdr reviewed (dismissed):
looks good to me considering the described plan of using a separate tag first, then change the v3 tag later once tests confirm it is working.

dskatz reviewed (changes requested):
The go.mod for drone helm 3 was not updated

lviana committed:
Update module versions

lviana commented:
> The go.mod for drone helm 3 was not updated

Thanks David, `go.mod` updated.

dskatz reviewed (commented):
There are a bunch of linting errors in addition to the go.mod update.

`make test`


This repo also does not look like it was fully moved over from mongodb-forks. For example; the go.mod file and package dependencies still reference.
`github.com/mongodb-forks/drone-helm3`


Also,  I am not sure why this is only being updated to 3.10.2.  This isn't even the latest patch release e.g. ([Helm v3.10.3](https://github.com/helm/helm/releases/tag/v3.10.3)).  

It seems like we should try upgrade helm as far as can be.

Commit:
Address linter violations

Added "nolint" tag to lines writing to stderr, since those are unlikely
to fail and are best effort at best for the cases handled

Commit:
Renamed module from "github.com/mongodb-forks/drone-helm3" to "github.com/kanopy-platform/drone-helm3"

Commit:
Bump helm to 3.10.3

Commit:
Ignore drone-helm binary file

Commit:
Downgrade helm.sh/helm to v3.10.2

Commit:
Update Dockerfile

Commit:
Upgrade dependencies to use helm v3.18.6

jaswdr commented:
As discussed offline in Slack, I'm upgrading to v3.18.6 instead of v3.10.2, as part of it I need to also to upgrade some dependencies, which you can see in the go.mod file.

jaswdr commented on build/Dockerfile-test:
```diff
@@ -1,4 +1,4 @@
-FROM golangci/golangci-lint:v1.48.0 as cache
+FROM golangci/golangci-lint:v2.5.0 as cache
```
required due to Go 1.25 bump

jaswdr commented on .drone.yml:
```diff
@@ -14,7 +14,7 @@ resources:
 steps:
   - name: test
-    image: golangci/golangci-lint:v1.56.0
+    image: golangci/golangci-lint:v2.5.0
```
required due to Go 1.25 upgrade, I'm also keeping this in sync with other repos

jaswdr commented on Dockerfile:
```diff
@@ -7,7 +7,7 @@ COPY . .
 
 # --- Copy the cli to an image with helm already installed ---
-FROM alpine/helm:3.8.1
+FROM alpine/helm:3.18.6
```
at the time of this PR, this is the latest v3.18.X version

jaswdr commented on go.mod:
```diff
@@ -1,151 +1,135 @@
 require (
 	github.com/golang/mock v1.6.0
-	github.com/helm/helm-2to3 v0.10.1
+	github.com/helm/helm-2to3 v0.11.0
```
It was required to bump this version because the older one uses a now removed OpenStack auth integration, which conflicts with helm v3.18.6

jaswdr commented on go.mod:
```diff
@@ -1,151 +1,135 @@
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
+	helm.sh/helm/v3 v3.18.6
+	k8s.io/api v0.33.3
```
this was upgraded because of helm.sh/helm v3.18.6 bump

dskatz replied:
If issues are discovered with using the latest version on k8s 1.33; that would be good justification for using an older version that closely matches our environments k8s version.  Its probably unlikely given that we have been using a super older version with no problem.

Commit:
Update sign of .drone.yml

richerve reviewed (approved):
LGTM

dskatz commented on internal/helm/plan.go:
```diff
@@ -82,6 +82,7 @@ func determineSteps(cfg env.Config) *func(env.Config) []Step {
 func (p *Plan) Execute() error {
 	for i, step := range p.steps {
 		if p.cfg.Debug {
+			//nolint:errcheck
```
This is fine; given that this is a fork and we are not the original authors and I am not interested in doing a refactor for proper error handling.

dskatz commented on go.mod:
```diff
@@ -1,151 +1,135 @@
-	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/docker/cli v20.10.11+incompatible // indirect
-	github.com/docker/distribution v2.7.1+incompatible // indirect
-	github.com/docker/docker v20.10.12+incompatible // indirect
-	github.com/docker/docker-credential-helpers v0.6.4 // indirect
```
This removal will resolve https://github.com/kanopy-platform/drone-helm3/security/dependabot/49

dskatz commented on go.mod:
```diff
@@ -1,151 +1,135 @@
+	github.com/xlab/treeprint v1.2.0 // indirect
+	go.yaml.in/yaml/v2 v2.4.2 // indirect
+	go.yaml.in/yaml/v3 v3.0.3 // indirect
+	golang.org/x/crypto v0.40.0 // indirect
```
This will resolve: https://github.com/kanopy-platform/drone-helm3/security/dependabot/52

dskatz reviewed (approved):
LGTM.

Our end to end testing should also validate helm 2 to 3 conversions.

jaswdr commented on go.mod:
```diff
@@ -1,151 +1,135 @@
+module github.com/kanopy-platform/drone-helm3
 
-go 1.21
+go 1.25
```
this was originally bumped to 1.24, but since we are using 1.25 in other places I bumped it to 1.25 instead.
